modules/vault-cluster/variables.tf

```diff
@@ -197,3 +197,32 @@ variable "force_destroy_s3_bucket" {
   description = "If 'configure_s3_backend' is enabled and you set this to true, when you run terraform destroy, this tells Terraform to delete all the objects in the S3 bucket used for backend storage. You should NOT set this to true in production or you risk losing all your data! This property is only here so automated tests of this module can clean up after themselves. Only used if 'enable_s3_backend' is set to true."
   default     = false
 }
+
+# Launch Template Extensions
+
+variable "asg_launch_mechanism" {
+  description = "Select between launch_config-driven or launch_template-driven autoscaling group."
+  default     = "launch_config"
```

Comment:
On line 14 of `main.tf` the expected value is `launch_configuration`, not `launch_config`. 

It's also worth adding here more clearly what are the expected values, or even maybe doing this in a different way, such as using booleans, to avoid confusion?

Another thing is that I think this is lacking an explanation of why you would choose one way over another. If not here, maybe a small note on the Readme?